doc/geany.txt

```diff
@@ -2674,6 +2674,29 @@ Placeholder   Description
               useful if you're debugging color schemes or related code.
 ============  ===========================================================
 
+
+Filetype Regex
+``````````````
+
+The ``extract_filetype_regex`` setting contains the regex that is used
+to extract the filetype name from the file with the first capture group.
+The default value finds filetypes contained in Emacs-style tags.
+See `def_ft_regex`_.
```

Comment:
Maybe we can add a title for the link here? Currently it looks like:
![Screenshot_2021-11-28_12-48-13](https://user-images.githubusercontent.com/617017/143766418-2899e666-7f76-4729-983b-e52edfd682d7.png)

Reply:
Changed to refer to the keyfile.c section.  Also put the default regex and link back to Filetype Regex directly in the table.